unstructured_inference/models/unstructuredmodel.py

```diff
@@ -0,0 +1,14 @@
+from abc import ABC, abstractmethod
+from typing import Any
+from PIL import Image
+
+
+class UnstructuredModel(ABC):
+    """Wrapper class for the various models used by unstructured."""
+
+    def __init__(self, model: Any):
+        self.model = model
+
+    @abstractmethod
+    def __call__(self, x: Image) -> Any:
```

Comment:
If we only accept Images, should this be called `UnstructuredImageModel`?

Reply:
not sure how i feel about objects that have the `__call__` method overridden, can make for hard to read code.
Is there any reason why the interface can't be `infer(...)`

Reply:
I guess there's no harm in making it more general here. Changed input type to `Any` and to `Image` in the detectron2 subclass.

Reply:
The `__call__` pattern for inference is what `transformers` does for `pipelines`. I actually don't mind that pattern here because it would be familiar to most of the people using this code.

Reply:
IMO a good pattern/name here would be something commonly used by other ML libraries for model inference.

I wouldn't be surprised if `infer` is used in an ML library, but I'm not aware of any.

I went with `__call__` since model objects you call for inference is a well established pattern found in torch and tensorflow/keras and propagates to libraries that wrap those such as fastai (although in each of these there are alternative ways to call the model for inference, `forward` in torch, `predict` in keras). Other common ones that I'm aware of are `forward` (didn't really like this because it suggests a feedforward process that's incidental to what we're doing here), `predict`/`predict_proba` (from `sklearn`), and most recently to me, `detect` from detectron. Of these I like `predict` the best. Maybe we could allow both `predict` and `__call__`.